src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs

```diff
@@ -164,7 +164,7 @@ internal static AuthorityType DetectAuthorityType(string authority)
             return authorityType;
         }
 
-        private static string CanonicalizeUri(string uri)
+        internal static string CanonicalizeUri(string uri)
```

Comment:
>internal static string CanonicalizeUri(string uri) [](start = 8, length = 50)

The URI class does canonicalization automatically, thus it should be considered using that (though not sure if that is available on .net core).

Reply:
Or perhaps the method really should be called EnsureUrlEndsWithForwardslash ...

---
In reply to: [157048639](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/911#discussion_r157048639) [](ancestors = 157048639)

Reply:
it is not general canonicalization of uri, It is authority specific canonicalization, So meant to be used only for Authority uri

Reply:
that's why I suggested to rename the method

Reply:
ok, will rename